query/src/users/auth/auth_mgr.rs

```diff
@@ -42,28 +42,37 @@ pub enum Credential {
 }
 
 impl AuthMgr {
-    pub async fn create(cfg: Config, users: Arc<UserApiProvider>) -> Result<Self> {
+    pub async fn create(cfg: Config, user_mgr: Arc<UserApiProvider>) -> Result<Self> {
         Ok(AuthMgr {
-            users,
+            user_mgr,
             tenant: cfg.query.tenant_id.clone(),
             jwt: JwtAuthenticator::try_create(cfg).await?,
         })
     }
 
     pub async fn no_auth(&self) -> Result<UserInfo> {
-        self.users
+        self.user_mgr
             .get_user(&self.tenant, UserIdentity::new("root", "127.0.0.1"))
             .await
     }
 
     pub async fn auth(&self, credential: &Credential) -> Result<UserInfo> {
         match credential {
             Credential::Jwt { token: t } => {
-                let user_name = match &self.jwt {
-                    Some(j) => j.get_user(t.as_str())?,
+                let jwt = match &self.jwt {
+                    Some(j) => j.get_jwt_claims(t.as_str())?,
                     None => return Err(ErrorCode::AuthenticateFailure("jwt auth not configured.")),
                 };
-                self.users
+                let user_name = jwt.subject.unwrap();
+                if let Some(create_user) = jwt.custom.create_user {
+                    let tenant = create_user.tenant_id.unwrap_or_else(|| self.tenant.clone());
+                    let mut user_info = UserInfo::new(&user_name, "%", AuthInfo::JWT);
```

Comment:
what to do with special names like "root"

Reply:
Should be dealt with with https://github.com/datafuselabs/databend/issues/4572 ?

Reply:
is root can be accessed with JWT? i guess it'd be safer to restrict the root access by %'127.0.0.1' q.q

Reply:
> i guess it'd be safer to restrict the root access by %'127.0.0.1'

Agree, so we should restrict the creation of root. see https://github.com/datafuselabs/databend/issues/4572

Reply:
> Should be dealt with with #4572 ?

I thought #4572 mean to restrict it in the parser.  Ok to deal with it later.